test/e2e/prometheus_test.go

```diff
@@ -264,7 +271,13 @@ func testMetricsFromPrometheusServer(t *testing.T, data *TestData, prometheusJob
 	// Target metadata API(/api/v1/targets/metadata) has been available since Prometheus v2.4.0.
 	// This API is still experimental in Prometheus v2.19.3.
 	path := url.PathEscape("match_target={job=\"" + prometheusJob + "\"}")
-	queryUrl := fmt.Sprintf("http://%s:%d/api/v1/targets/metadata?%s", hostIP, nodePort, path)
+	var ipAndPort string
+	if net.ParseIP(hostIP).To4() != nil {
```

Comment:
Nit: this could be a func

Reply:
Good idea. Use `net.JoinHostPort` instead.